src/main/java/org/mskcc/cmo/metadb/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -0,0 +1,271 @@
+                String newSamplesString = mapper.writeValueAsString(validSampleList.toArray(new Object[0]));
+                requestJsonMap.replace("samples", newSamplesString);
+                LOG.info("CMO request passed sanity checking - missing samples...");
+                requestStatusLogger.logRequestStatus(requestJson,
+                        RequestStatusLogger.StatusType.CMO_REQUEST_MISSING_REQ_FIELDS);
+            }
+            return mapper.writeValueAsString(requestJsonMap);
+        }
+        LOG.error("Request failed sanity checking - logging request status...");
+        requestStatusLogger.logRequestStatus(requestJson,
+                RequestStatusLogger.StatusType.CMO_REQUEST_FAILED_SANITY_CHECK);
+        return null;
+    }
+
+    /**
+     * isValidCmoSample receives a sampleMap
+     * 
+     * <p><ul>
+     * <li>Checks if the sample has all the required fields
+     * <li>If RequestId, InvestigatorSampleId, CmoPatientId,
+     * SpecimenType, SampleType or NormalizedPatientId are missing, returns false
+     * </ul>
+     * @throws JsonProcessingException or JsonMappingException 
+     */
+    private boolean isValidCmoSample(Map<String, String> sampleMap,
+            boolean isCmoRequest, boolean hasRequestId) throws JsonMappingException, JsonProcessingException {
+        if (sampleMap.isEmpty() || sampleMap == null) {
+            return Boolean.FALSE;
+        }
+        if (isCmoRequest && !hasBaitSet(sampleMap)) {
+            return Boolean.FALSE;
+        }
+        if (!hasRequestId || !hasInvestigatorSampleId(sampleMap)
+                || !hasCmoPatientId(sampleMap) || !hasSpecimenType(sampleMap)
+                || !hasSampleType(sampleMap) || !hasNormalizedPatientId(sampleMap)) {
+            return Boolean.FALSE;
+        }
+        return Boolean.TRUE;
+    }
+
+    /**
+     * isValidNonCmoSample receives a sampleMap
+     * 
+     * <p><ul>
+     * <li>Check if the sample has all the required fields
+     * <li>If baitSet or normalizedPatientId is missing, returns false
+     * </ul>
+     * @throws JsonProcessingException or JsonMappingException 
+     */
+    private boolean isValidNonCmoSample(Map<String, String> sampleMap)
+            throws JsonMappingException, JsonProcessingException {
+        if (sampleMap.isEmpty()
+                || sampleMap == null
+                || !hasBaitSet(sampleMap)
+                || !hasNormalizedPatientId(sampleMap)) {
+            return Boolean.FALSE;
+        }
+        return Boolean.TRUE;
+    }
+
+    private boolean isCmoRequest(Map<String, Object> requestJsonMap) {
+        if (Strings.isBlank(requestJsonMap.get("cmoRequest").toString())) {
+            return Boolean.FALSE;
+        }
+        return Boolean.valueOf(requestJsonMap.get("cmoRequest").toString());
+    }
+
+    private boolean hasRequestId(Map<String, Object> requestJsonMap) {
+        return Strings.isBlank(requestJsonMap.get("requestId").toString());
```

Comment:
so similar comment here. 

if requestJsonMap.get("requestId") returns null and you try to invoke `.toString()` on it then we'll get a null pointer exception

These checks should be changed to something like 

return (requestJsonMap.containsKey("requestId") && !Strings.isBlank(requestJsonMap.get("requestId")))

Reply:
!Strings.isBlank() wouldn't work right, cause requestJsonMap.get("requestId") has to be a string. Maybe something like this: return (requestJsonMap.get("requestId") != null && !Strings.isBlank(requestJsonMap.get("requestId").toString()));